src/ipmi.c

```diff
@@ -682,166 +815,410 @@ static void domain_connection_change_handler(ipmi_domain_t *domain, int err,
                                              unsigned int port_num,
                                              int still_connected,
                                              void *user_data) {
-  int status;
 
   DEBUG("domain_connection_change_handler (domain = %p, err = %i, "
         "conn_num = %u, port_num = %u, still_connected = %i, "
-        "user_data = %p);\n",
+        "user_data = %p);",
         (void *)domain, err, conn_num, port_num, still_connected, user_data);
 
-  status = ipmi_domain_add_entity_update_handler(
-      domain, domain_entity_update_handler, /* user data = */ NULL);
+  if (err != 0)
+    c_ipmi_error("domain_connection_change_handler", err);
+
+  if (!still_connected)
+    return;
```

Comment:
would a simple info/debug message not be useful here?

Reply:
At practice, that will be reported by a `c_ipmi_error()` call at several lines above.
Also, that will be reported by `c_ipmi_log()` handler too.

An example:
```
[2017-10-08 19:09:49] ipmi plugin: one 0 ipmi_lan.c(lost_connection): Connection 0 to the BMC is down
[2017-10-08 19:09:49] ipmi plugin: one 0 ipmi_lan.c(lost_connection): All connections to the BMC are down
[2017-10-08 19:09:49] ipmi plugin: domain_connection_change_handler failed for `one`: OS: Connection timed out
```

I have added a code to send Collectd notification in case of connection lost/restored and will publish it in updated PR.